### ISSUE TITLE
ValueError: y has more than n_class unique elements.

### ISSUE THREAD

**Issue by BlakeList** [deepchem/deepchem#2474]
## 🐛 Bug

Hi there,

Not sure if I am doing something wrong, but the code for the multiclass tutorial 
[multiclass_sklearn.py](https://github.com/deepchem/deepchem/blob/master/examples/multiclass/multiclass_sklearn.py)

Creates an error:

```
ValueError                                Traceback (most recent call last)
<ipython-input-153-2de1211744a4> in <module>
     20 
     21 print("About to evaluate model")
---> 22 train_scores = model.evaluate(dataset, sklearn.metrics.roc_auc_score, [])
     23 
     24 print("Train scores")

/workspace/appscratch/miniconda/cflbwl_black/lib/python3.7/site-packages/deepchem/models/models.py in evaluate(self, dataset, metrics, transformers, per_task_metrics, use_sample_weights, n_classes)
    218         per_task_metrics=per_task_metrics,
    219         use_sample_weights=use_sample_weights,
--> 220         n_classes=n_classes)
    221 
    222   def get_task_type(self) -> str:

/workspace/appscratch/miniconda/cflbwl_black/lib/python3.7/site-packages/deepchem/utils/evaluate.py in compute_model_performance(self, metrics, csv_out, stats_out, per_task_metrics, use_sample_weights, n_classes)
    314           n_tasks=n_tasks,
    315           n_classes=n_classes,
--> 316           use_sample_weights=use_sample_weights)
    317       if per_task_metrics:
    318         multitask_scores[metric.name], computed_metrics = results

/workspace/appscratch/miniconda/cflbwl_black/lib/python3.7/site-packages/deepchem/metrics/metric.py in compute_metric(self, y_true, y_pred, w, n_tasks, n_classes, per_task_metrics, use_sample_weights, **kwargs)
    617 
    618     y_true = normalize_labels_shape(
--> 619         y_true, mode=self.mode, n_tasks=n_tasks, n_classes=n_classes)
    620     y_pred = normalize_prediction_shape(
    621         y_pred, mode=self.mode, n_tasks=n_tasks, n_classes=n_classes)

/workspace/appscratch/miniconda/cflbwl_black/lib/python3.7/site-packages/deepchem/metrics/metric.py in normalize_labels_shape(y, mode, n_tasks, n_classes)
    177       # check whether n_classes is int or not
    178       assert isinstance(n_classes, int)
--> 179       y_hot = to_one_hot(y_task, n_classes=n_classes)
    180       y_hot = np.expand_dims(y_hot, 1)
    181       all_y_task.append(y_hot)

/workspace/appscratch/miniconda/cflbwl_black/lib/python3.7/site-packages/deepchem/metrics/metric.py in to_one_hot(y, n_classes)
    398     raise ValueError("y must be a vector of shape (N,) or (N, 1)")
    399   if len(np.unique(y)) > n_classes:
--> 400     raise ValueError("y has more than n_class unique elements.")
    401   N = np.shape(y)[0]
    402   y_hot = np.zeros((N, n_classes))

ValueError: y has more than n_class unique elements.
```

## Environment

* OS:
* Python version: 3.7.4
* DeepChem version: 2.5.0
* RDKit version (optional): 2021.03.1
* TensorFlow version (optional): 2.4.1
* PyTorch version (optional): 1.7.1

**Comment by rbharath** [comment]
Hmm, not sure what's going on. Will mark this as a bug and good first issue to investigate

**Comment by peterskipper** [comment]
I can take this one on, if no one else has already.

**Comment by rbharath** [comment]
@peterskipper Yes, feel free to go for it! This could perhaps just be a bug that crept into the example but I'm not sure

**Comment by rbharath** [comment]
Fixed by #2506
